pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -499,9 +507,8 @@ func (a *FlavorAssigner) assignFlavors(log logr.Logger, counts []int32) Assignme
 				break
 			}
 		}
-		var groupStatus *Status
-
-		for resName := range requests {
+		var groupStatus Status
+		for _, resName := range slices.Sorted(maps.Keys(requests)) {
```

Comment:
Without the sort on test started to flake because the messages were in different order, depending on the order of the keys in the map: https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_kueue/6119/pull-kueue-test-unit-main/1947299916032249856

I'm not sure why it didn't flake in the past, but it is probably better to explicitly order the resources.

Reply:
Ok, I figured it out, this is because the *Status implemented Equal function, which was actually used in tests. Since I removed the ptr the function started to not be used. 

Since we don't need this function in production code I moved its role to statusComparer, it also makes it much easier to find as it is directly referenced from cmp.Diff invocation. Finally, the test code does not pollute the prod code.

Still, the randomness in the messages may be surprising, but let's keep it out of scope for the PR.

Reply:
I like the idea of sorting the messages while settings instead of getting